File: mrcomic-theme-platform/src/models/Theme.js
const mongoose = require('mongoose');

const themeSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    trim: true,
    minlength: 3,
    maxlength: 50
  },
  description: {
    type: String,
    required: true,
    trim: true,
    maxlength: 500
  },
  version: {
    type: String,
    required: true,
    default: '1.0.0'
  },
  isPublic: {
    type: Boolean,
    default: true
  },
  previewImageUrl: {
    type: String,
    required: true
  },
  authorId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  rating: {
    average: {
      type: Number,
      default: 0
    },
    count: {
      type: Number,
      default: 0
    }
  },
  downloads: {
    type: Number,
    default: 0
  },
  themeData: {
    colors: {
      primary: String,
      secondary: String,
      accent: String,
      background: String,
      text: String
    },
    fonts: {
      main: String,
      headers: String,
      size: String
    },
    elements: {
      cornerRadius: Number,
      iconStyle: String,
      buttonStyle: String
    }
  },
  tags: [{
    type: String,
    trim: true
  }]
}, {
  timestamps: true
});

// Индексы для быстрого поиска
themeSchema.index({ name: 'text', description: 'text', tags: 'text' });
themeSchema.index({ authorId: 1 });
themeSchema.index({ 'rating.average': -1 });
themeSchema.index({ downloads: -1 });
themeSchema.index({ createdAt: -1 });

const Theme = mongoose.model('Theme', themeSchema);

module.exports = Theme;

File: app/src/main/java/com/mrcomic/theme/data/model/Theme.kt
package com.example.mrcomic.theme.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName

/**
 * Модель данных для темы
 */
@Entity(tableName = "themes")
data class Theme(
    @PrimaryKey
    @SerializedName("id")
    val id: String,
    
    @SerializedName("name")
    val name: String,
    
    @SerializedName("description")
    val description: String,
    
    @SerializedName("version")
    val version: String,
    
    @SerializedName("isPublic")
    val isPublic: Boolean,
    
    @SerializedName("previewImageUrl")
    val previewImageUrl: String,
    
    @SerializedName("authorId")
    val authorId: String,
    
    @SerializedName("author")
    val author: User?,
    
    @SerializedName("rating")
    val rating: ThemeRating,
    
    @SerializedName("downloads")
    val downloads: Int,
    
    @SerializedName("createdAt")
    val createdAt: String,
    
    @SerializedName("updatedAt")
    val updatedAt: String,
    
    @SerializedName("themeData")
    val themeData: ThemeData,
    
    @SerializedName("tags")
    val tags: List<String>
)

data class ThemeRating(
    @SerializedName("average")
    val average: Float,
    
    @SerializedName("count")
    val count: Int
)

data class ThemeData(
    @SerializedName("colors")
    val colors: ThemeColors,
    
    @SerializedName("fonts")
    val fonts: ThemeFonts,
    
    @SerializedName("elements")
    val elements: ThemeElements
)

data class ThemeColors(
    @SerializedName("primary")
    val primary: String,
    
    @SerializedName("secondary")
    val secondary: String,
    
    @SerializedName("accent")
    val accent: String,
    
    @SerializedName("background")
    val background: String,
    
    @SerializedName("text")
    val text: String
)

data class ThemeFonts(
    @SerializedName("main")
    val main: String,
    
    @SerializedName("headers")
    val headers: String,
    
    @SerializedName("size")
    val size: String
)

data class ThemeElements(
    @SerializedName("cornerRadius")
    val cornerRadius: Int,
    
    @SerializedName("iconStyle")
    val iconStyle: String,
    
    @SerializedName("buttonStyle")
    val buttonStyle: String
)
